### PR TITLE
Cmr 4320

### PR DESCRIPTION
**Pull request by laurenfrederick**
Added validation to give a warning when the ends at present flag is true and the end date is set on a collection.

Moved temporal extent validation and tests into separate files.

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj [hunk 1]
```diff
@@ -0,0 +1,42 @@
+(ns cmr.umm-spec.validation.temporal-extent
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj]: 👍

File: umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    {field-path [(str "Ending date should be in the past. Either set ending date to a date "
+                      "in the past or remove end date and set the ends at present flag to true.")]}))
+
+(defn- ends-at-present-validation
+  "Validate that the collection ends at present flag and end date time are
+  not both set"
+  [field-path value]
+  (when (and (= true (:EndsAtPresentFlag value))
+             (seq (:RangeDateTimes value))
+             (not (some nil? (map :EndingDateTime (:RangeDateTimes value)))))
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj]: Nit: Maybe easier to read as:
```(seq (keep :EndingDateTime (:RangeDateTimes value)))```
I just have trouble with double-negatives I think, nothing really wrong with the code.

File: umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {field-path [(str "Ends at present flag is set to true, but an ending date is "
+                      "specified. Remove the latest ending date or set the ends at "
+                      "present flag to false.")]}))
+
+(def temporal-extent-warning-validation
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj]: Missing docstrings for the next two vars.

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj [hunk 4]
```diff
@@ -63,24 +51,12 @@
...
-                             :EndingDateTime temporal-end-date-in-past-validator})
-   :SingleDateTimes (v/every vu/date-in-past-validator)})
-
 (def collection-validation-warnings
  "Defines validations for collections that we want to return as warnings and not
  as failures"
  {:RelatedUrls (v/every url/related-url-validations)
   :Projects project/projects-warning-validation
-  :TemporalExtents (v/every temporal-extent-warning-validation)
+  :TemporalExtents (v/every temporal-extent/temporal-extent-warning-validation)
```
Comment (chris-durbin) [inline umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj]: If you pass in the strict validation header do we return failures if there is any warning? I don't see a test for it, which we should probably add if that's how it works.

Comment (laurenfrederick) [reply umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj]: Are you talking about the validate-umm-c header? If so, there is a test in collection-umm-spec-validation-test (cmr.system-int-test.ingest.collection-umm-spec-validation-test) that tests that the warnings are returned as errors when the header is true.

File: umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj (hunk 2 above)
Comment (laurenfrederick) [reply umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj]: I don't think that is the same logic? I am looking to make sure that there is at least one date range with an ending date time of nil.

File: umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj (hunk 3 above)
Comment (laurenfrederick) [reply umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj]: Fixed, thanks.

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj (hunk 4 above)
Comment (chris-durbin) [reply umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj]: I think that's the header - so this warning would be treated as an error in that case? I'm ok with it if there's existing tests, I just wasn't sure of the behavior.

File: umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj (hunk 2 above)
Comment (chris-durbin) [reply umm-spec-lib/src/cmr/umm_spec/validation/temporal_extent.clj]: Oh, I retract my comment and code.

File: umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj (hunk 4 above)
Comment (laurenfrederick) [reply umm-spec-lib/src/cmr/umm_spec/validation/umm_spec_collection_validation.clj]: Yes, in that case the warning would be treated as an error. There is a test for the header, but not specific to every validation warning.

**Pull request merged**